diagnostics/collect-networking-logs.ps1

```diff
@@ -105,9 +122,16 @@ try
           183  # Application 2
 
     $Key = $null
-    while ($Key -Eq $null -Or $Key.VirtualKeyCode -Eq $null -Or $KeysToIgnore -Contains $Key.VirtualKeyCode)
+    while (($Key -Eq $null -Or $Key.VirtualKeyCode -Eq $null -Or $KeysToIgnore -Contains $Key.VirtualKeyCode) -and ($null -eq $tcpdumpProcess -or $tcpdumpProcess.HasExited -eq $false))

```

Comment:
right now the tcpdump process will not exit on its own, it will be stopped only when the user presses a key, so we can't use this as an indication to stop the script

Reply:
Here I am trying to introduce an alternative mechanism to end log collecting. Instead of having the user press a key, we can have a test automation script kill process _tcpdump_ which will be detected here and will effectively end the log collection.
This does not change the behavior of _collect-networking-logs.ps1_ since the very next thing it does is to terminate _tcpdump_ anyway.

Reply:
I see, thanks for clarifying

the WSL distro might not have tcpdump installed. In the instructions we have for running the script, we tell users to manually install it (https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#collect-wsl-logs-for-networking-issues)

if they don't have installed, we'll exit the script too early because $tcpdumpProcess will be null

can we find a different way to automatically stop the script?

Reply:
sorry, I missed that the change already takes into account that $tcpdumpProcess can be null

Reply:
this looks good in this case, I am resolving the comment